app/scripts/models/account.js

```diff
@@ -16,31 +16,41 @@ define([
   'lib/constants',
   'models/profile-image'
 ], function (Backbone, _, p, AuthErrors, ProfileClient, Constants, ProfileImage) {
-
   // Account attributes that can be persisted
   var PERSISTENT = {
-    uid: undefined,
+    accessToken: undefined,
     email: undefined,
+    grantedPermissions: undefined,
+    lastLogin: undefined,
+    needsOptedInToMarketingEmail: undefined,
+    profileImageId: undefined,
+    profileImageUrl: undefined,
     sessionToken: undefined,
     sessionTokenContext: undefined,
-    accessToken: undefined,
-    verified: undefined,
-    profileImageUrl: undefined,
-    profileImageId: undefined,
-    lastLogin: undefined,
-    grantedPermissions: undefined
+    uid: undefined,
+    verified: undefined
   };
 
   var DEFAULTS = _.extend({
-    password: undefined,
-    unwrapBKey: undefined,
+    customizeSync: undefined,
     keyFetchToken: undefined,
-    customizeSync: undefined
+    password: undefined,
+    unwrapBKey: undefined
   }, PERSISTENT);
 
   var ALLOWED_KEYS = Object.keys(DEFAULTS);
   var ALLOWED_PERSISTENT_KEYS = Object.keys(PERSISTENT);
 
+  function withMarketingEmailClient(method) {
+    return function () {
+      var self = this;
+      return self.getOAuthTokenFromSessionToken(self.get('sessionToken'), 'basket')
```

Comment:
@rfk - is this correct?

I've been waffling whether email marketing related functionality should live here and an email-marketing-client is passed in, or whether the email marketing client should be passed to the views and its functions accept an account. This seems like the the more natural of the two because it means account callers do not have to worry about the email marketing client, but I could be convinced otherwise.